logstash-core/lib/logstash/pipeline_action/stop.rb

```diff
@@ -12,13 +12,19 @@ def initialize(pipeline_id)
     end
 
     def execute(agent, pipelines)
-      pipeline = pipelines[pipeline_id]
-      pipeline.shutdown { LogStash::ShutdownWatcher.start(pipeline) }
-      pipeline.thread.join
-      pipelines.delete(pipeline_id)
+      pipelines.compute(pipeline_id) do |k,pipeline|
```

Comment:
I believe, given your comment, that there's nothing to do here?

One point btw: all the `CHM#compute*` functions return the new value, so the second example wouldn't work. The code as is is buggy as you mention below, since this should always return nil.